cirq-core/cirq/transformers/transformer_primitives_test.py

```diff
@@ -810,3 +810,20 @@ def wrapped_merge_func(op1, op2):
         _ = cirq.merge_operations(circuit, wrapped_merge_func)
         total_operations = len([*circuit.all_operations()])
         assert wrapped_merge_func.num_function_calls <= 2 * total_operations
+
+
+def test_merge_operations_does_not_merge_ccos_behind_measurements():
+    q = cirq.LineQubit.range(2)
+    cco_op = cirq.X(q[1]).with_classical_controls("a")
+
+    def merge_func(op1, op2):
+        return cirq.I(*op1.qubits) if op1 == cco_op and op2 == cco_op else None
+
+    circuit = cirq.Circuit([cirq.H(q[0]), cirq.measure(q[0], key="a"), cco_op] * 2)
+    cirq.testing.assert_same_circuits(cirq.merge_operations(circuit, merge_func), circuit)
+
+    circuit = cirq.Circuit([cirq.H(q[0]), cirq.measure(q[0], key="a"), cco_op, cco_op] * 2)
```

Comment:
This doesn't seem like it achieves the test's stated purpose. The two CCOs in the first half get merged and the two in the second half get merged, so AFAICT the don't-merge-across-conflicting-measure check isn't tested.

Reply:
The `don't-merge-across-conflicting-measure` is checked in the first assertion on line 823. The second assertion is to make sure that mergeable cco's actually get merged.